### PR TITLE
Fix API documentation generation

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
* fix issues with jsdocs causing the binary to exit with a failure
* update the vjsstandard to 6.0.0 to support jsdoc validation
* fix some minor jsdoc style issues

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/setup.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
 /**
- * Functions for setting up a player without user insteraction based on the data-setup
- * `attribute` of the video tag.
+ * @file setup.js - Functions for setting up a player without
+ * user insteraction based on the data-setup `attribute` of the video tag.
```
Comment (misteroneill) [inline src/js/setup.js]: Might as well fix the "insteraction" typo here.

File: src/js/tech/flash-rtmp.js [hunk 2]
```diff
@@ -104,8 +104,7 @@ function FlashRtmpDecorator(Flash) {
   /**
    * Regular expression used to check if the source is an rtmp source.
    *
-   * @property
-   * @type {RegExp}
+   * @property {RegExp} Flash.RTMP_RE
```
Comment (misteroneill) [inline src/js/tech/flash-rtmp.js]: I believe this could simply be `@type {RegExp}` without the `@property` entirely.

File: src/js/utils/events.js [hunk 3]
```diff
@@ -1,11 +1,10 @@
 /**
- * @file events.js
- * @module events
- *
- * Event System (John Resig - Secrets of a JS Ninja http://jsninja.com/)
+ * @file events.js. An Event System (John Resig - Secrets of a JS Ninja http://jsninja.com/)
```
Comment (misteroneill) [inline src/js/utils/events.js]: Ahhhh, I was wondering what we could do about this. I had a less appealing fix. 👍

File: src/js/tech/flash-rtmp.js (hunk 2 above)
Comment (BrandonOCasey) [inline src/js/tech/flash-rtmp.js]: It could be we probably want to specify that it is added to the Flash tech.

**Comment by BrandonOCasey** [comment]
@misteroneill updated this PR to depend on #3841 and moved examples in the jsdoc to guides

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -245,27 +244,25 @@ Follow these guidelines:
...
-fix: one line commit explanation
+An example of the first line of a commit message: `fix: changed the footer to correctly display foo`
 
 In the body of the commit message, we can talk about why we made the change. What the change entails.
+Any testing considerations or things to think about when looking at the commit. For Example:
 
-Any testing considerations or things to think about when looking at the commit.
-
-Fixes #123. The footer can contain Fixes messages.
+```txt
```
Comment (gkatsev) [inline CONTRIBUTING.md]: this still doesn't match the original version.

File: docs/guides/components.md [hunk 5]
```diff
@@ -55,12 +110,180 @@ Results in a DOM that looks like this:
 </div>
 ```
 
-Again, refer to [the component API docs](http://docs.videojs.com/docs/api/component.html) for complete details on methods available for managing component structures.
+### Using Opitons
```
Comment (gkatsev) [inline docs/guides/components.md]: typo, opitons->options

Comment (gkatsev) [inline docs/guides/components.md]: Seems like this is a new section altogether. Can we pull additions into a separate PR? Making this PR just adding in remark and the changes it makes.

**Comment by gkatsev** [comment]
And looking more into this I can see one issue is that this PR is based on https://github.com/videojs/video.js/pull/3841.

File: docs/guides/components.md [hunk 6]
```diff
@@ -36,8 +43,42 @@ In addition, there are a couple methods worth recognizing:
...
+```js
+// adding a button to the player
+var player = videojs('some-video-id');
+var Component = videojs.getComponent('Component');
+var button = new Component(player);
+
+console.log(button.el());
+```
+
+The Above code will output
```
Comment (misteroneill) [inline docs/guides/components.md]: "Above" shouldn't be capitalized.

File: docs/guides/events.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
+# Events
```
Comment (gkatsev) [inline docs/guides/events.md]: Is this for EventTarget or events.js? They're not 100% equivalent.

Comment (gkatsev) [inline docs/guides/events.md]: Sounds like this is for EventTarget, so, we should rename the file and this title to be Event Target.

File: docs/guides/components.md [hunk 8]
```diff
@@ -69,7 +110,174 @@ Results in a DOM that looks like this:
...
+var myComponent = new Component(player);
+var myButton = myComponent.addChild('MyButton', {
+  text: 'Press Me',
+  buttonChildExample: {
+    buttonChildOption: true
+  }
+});
+```
+
+Childern can also be added via options when a component is intialized.
```
Comment (misteroneill) [inline docs/guides/components.md]: Two typos here: "Childern" should be "Children" and "intialized" should be "initialized".

File: docs/guides/components.md [hunk 9]
```diff
@@ -69,7 +110,174 @@ Results in a DOM that looks like this:
...
+  }, {
+    name: 'button',
+    someOtherOption: false
+  }]
+});
+```
+
+## Event Listening
+
+### using `on`
```
Comment (misteroneill) [inline docs/guides/components.md]: "using" should be capitalized.

File: docs/guides/player-workflows.md [hunk 10]
```diff
@@ -77,6 +89,275 @@ modal.on('hide', function() {
...
+  myPlayer.src("http://www.example.com/path/to/video.mp4");
+  myPlayer.ready(function() {
+    // get
+    var howLoudIsIt = myPlayer.volume();
+    // set
+    myPlayer.volume(0.5); // Set volume to half
+  });
+```
+
+Volume can also be muted (without actually changing the volume value) using the `muted` funciton. Calling it without an argument will return the current status of muted on the player.
```
Comment (gkatsev) [inline docs/guides/player-workflows.md]: `funciton` -> `function`.

File: docs/guides/components.md [hunk 11]
```diff
@@ -69,7 +110,174 @@ Results in a DOM that looks like this:
...
+myComponent.on('eventType', myFunc);
+myComponent.trigger('eventType');
+// logs 'myFunc called'
+
+myComponent.off('eventType', myFunc);
+myComponent.trigger('eventType');
+// does nothing
+```
+
+If myFunc gets excluded, ALL listeners for the event type will get removed. If
```
Comment (misteroneill) [inline docs/guides/components.md]: Instead of all caps here, maybe **bold**? Also on the line below.

File: docs/guides/events.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+* [on() || addEventListener()](#on--addeventlistener)
+* [off() || removeEventListener](#off--removeeventlistener)
+* [one()](#one)
+* [trigger() || dispatchEvent](#trigger--dispatchevent)
+
+## Overview
+
+Events in video.js are setup so that they mimic the DOM API that is used on object, but also have helpful shorthand functions with the same functionality.
+
+## `on()` \|\| `addEventListener()`
```
Comment (misteroneill) [inline docs/guides/events.md]: Why "||" instead of "or"? I think we should stick to English here.

File: docs/guides/events.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+  foo.on('bar', handleBar);
+
+  // This causes any `event listeners` for the `bar` event to get called
+  // see {@link EventTarget#trigger} for more information
+  foo.trigger('bar');
+  // logs 'bar was triggered'
+```
+
+## `off()` \|\| `removeEventListener`
```
Comment (misteroneill) [inline docs/guides/events.md]: Same comment about "||".

File: docs/guides/player-workflows.md [hunk 14]
```diff
@@ -77,6 +89,275 @@ modal.on('hide', function() {
...
+
+  // set
+  myPlayer.poster('http://example.com/myImage.jpg');
+
+  // get
+  console.log(myPlayer.poster());
+  // 'http://example.com/myImage.jpg'
+```
+
+## Accesing the Tech on the player
```
Comment (gkatsev) [inline docs/guides/player-workflows.md]: This reminds me, I'm thinking of changing this in 6.0. I want just `tech()` to log a warning along the lines of "I sure hope you know what you're doing" unless the object (or anything) is passes as an argument to `tech()`. It will always return `tech` regardless, it will just log the warning if no argument was passed.

File: docs/guides/videojs.md [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+  var someClass = function() {};
+  var someObj = new someClass();
+  videojs.bind(someObj, function() {
+    // this will be the context of someObj here
+  });
+```
+
+## `plugin()`
+
+**See the plugin guide in the docs for a more detailed example**
```
Comment (gkatsev) [inline docs/guides/videojs.md]: We should add a link to the doc if it exists currently.

File: docs/guides/videojs.md [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+
+## `getComponent()`
+
+```js
+  var VjsButton = videojs.getComponent('Button');
+  // Create a new instance of the component
+  var myButton = new VjsButton(myPlayer);
+```
+
+## `registerComponent`
```
Comment (misteroneill) [inline docs/guides/videojs.md]: For consistency with others, this should probably have `()` in there.

File: package.json [hunk 17]
```diff
@@ -115,7 +115,7 @@
     "time-grunt": "^1.1.1",
     "uglify-js": "~2.7.3",
     "videojs-doc-generator": "0.0.1",
-    "videojs-standard": "^5.2.0",
+    "videojs-standard": "^6.0.1",
```
Comment (gkatsev) [inline package.json]: This should probably be in a separate PR

File: docs/guides/events.md (hunk 12 above)
Comment (gkatsev) [inline docs/guides/events.md]: I changed this to `and`.

File: docs/guides/event-target.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+foo.on('bar', handleBar);
+
+// This causes any `event listeners` for the `bar` event to get called
+// see {@link EventTarget#trigger} for more information
+foo.trigger('bar');
+// logs 'bar was triggered'
+```
+
+## `off()` and `removeEventListener`
```
Comment (misteroneill) [inline docs/guides/event-target.md]: Missing parens after `removeEventListener`.

File: docs/guides/event-target.md [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+// removed after the first trigger
+foo.one('bar', handleBar);
+foo.trigger('bar');
+// logs 'bar was triggered'
+
+foo.trigger('bar');
+// does nothing
+```
+
+## `trigger()` and `dispatchEvent`
```
Comment (misteroneill) [inline docs/guides/event-target.md]: Missing parens after `dispatchEvent`.

File: docs/guides/videojs.md (hunk 15 above)
Comment (misteroneill) [inline docs/guides/videojs.md]: It does exist.

File: package.json (hunk 17 above)
Comment (misteroneill) [inline package.json]: Agreed.

File: docs/guides/videojs.md (hunk 15 above)
Comment (gkatsev) [inline docs/guides/videojs.md]: Oops, forgot to add the link.
